python-avd/pyavd/_eos_designs/structured_config/network_services/utils.py

```diff
@@ -341,7 +341,7 @@ def get_vrf_rt(self: AvdStructuredConfigNetworkServices, vrf: EosDesigns._Dynami
         """Return a string with the route-target for one VRF."""
         rt_override = vrf.rt_override
 
-        if rt_override is not None and ":" in rt_override:
+        if isinstance(rt_override, str) and (":" in rt_override or rt_override == "auto"):
```

Comment:
The check for None is much "cheaper" than isinstance, so when the type can only be None or str, we prefer that.
```suggestion
        if rt_override is not None and (":" in rt_override or rt_override == "auto"):
```